src/renderer/services/slippi/slippi.service.ts

```diff
@@ -25,45 +29,53 @@ const handleErrors = (errors: readonly GraphQLError[] | undefined) => {
 };
 
 class SlippiBackendClient implements SlippiBackendService {
-  private httpLink: HttpLink;
   private client: ApolloClient<NormalizedCacheObject>;
 
   constructor(
     private readonly authService: AuthService,
     private readonly dolphinService: DolphinService,
     clientVersion?: string,
   ) {
-    this.httpLink = new HttpLink({ uri: SLIPPI_BACKEND_URL });
-    this.client = new ApolloClient({
-      link: this.httpLink,
-      cache: new InMemoryCache(),
-      name: "slippi-launcher",
-      version: clientVersion,
-    });
-  }
+    const httpLink = new HttpLink({ uri: SLIPPI_BACKEND_URL });
+    const authLink = setContext(async () => {
+      // The firebase ID token expires after 1 hour so we will update the header on all actions
+      const token = await this.authService.getUserToken();
```

Comment:
Since the constructor is now getting pretty complicated, can you put this client creation logic into a private function such that our constructor can simply be something like:
```
this.client = this._initApolloClient(clientVersion);
```